### PR TITLE
abcl: bump version

### PR DESCRIPTION
**Pull request by hoanga**
this updates abcl to the latest release

note that an updated jdk (17+) is recommended per release notes

```
> abcl
Armed Bear Common Lisp 1.9.2
Java 17.0.7 N/A
OpenJDK 64-Bit Server VM
Low-level initialization completed in 0.308 seconds.
Startup completed in 1.714 seconds.
Type ":help" for a list of available commands.
CL-USER(1):
```

### CODE REVIEW

File: dev-java/abcl/abcl-1.9.2.recipe [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+BUILD_REQUIRES="
+	"
+BUILD_PREREQUIRES="
+	java:environment == 17
+	"
+
+INSTALL()
+{
+	source /system/data/profile.d/openjdk17.sh
+	export JAVA_HOME=$JDK17_HOME
```
Comment (korli) [inline dev-java/abcl/abcl-1.9.2.recipe:36]: Why is this needed?

File: dev-java/abcl/abcl-1.9.2.recipe [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	"
+REQUIRES="
+	haiku
+	java:environment >= 17
+	"
+
+BUILD_REQUIRES="
+	"
+BUILD_PREREQUIRES="
+	java:environment == 17
```
Comment (korli) [inline dev-java/abcl/abcl-1.9.2.recipe:30]: Why is this needed?

**Comment by Begasus** [comment]
Please also remove the old version while you're in the process.

File: dev-java/abcl/abcl-1.9.2.recipe (hunk 1 above)
Comment (hoanga) [reply dev-java/abcl/abcl-1.9.2.recipe]: i am not entirely sure. it came from the earlier port definitions where i might have ran into some circular conditions that don't appear to exist anymore.  have removed this. thanks!

File: dev-java/abcl/abcl-1.9.2.recipe (hunk 2 above)
Comment (hoanga) [reply dev-java/abcl/abcl-1.9.2.recipe]: similar to above and have had it removed
